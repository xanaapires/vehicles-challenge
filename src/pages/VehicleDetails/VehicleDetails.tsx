import React, { useMemo } from "react";
import Grid from "@mui/material/Grid";
import { useVehicles } from "../../hooks/useVehicles";
import { useParams } from "react-router-dom";
import { Box, Button, IconButton, Typography } from "@mui/material";
import ArrowBackIcon from "@mui/icons-material/ArrowBack";
import StarBorderOutlinedIcon from "@mui/icons-material/StarBorderOutlined";
import StarOutlinedIcon from "@mui/icons-material/StarOutlined";
import emptyPhoto from "../../assets/empty_photo.jpg";
import moment from "moment";
import { getRelativeTime } from "../../utils/dates";
import { useNavigate } from "react-router-dom";
import { defaultColors } from "../../components/ThemeProvider/themeConfigs";
import { Accordion } from "../../components/Accordion";
import { LabelledText } from "../../components/LabelledText";
import { ROUTES } from "../../utils/routes";

const VehicleDetails = () => {
  const navigate = useNavigate();
  const { vehicleId } = useParams();
  const { getVehicleById, toggleFavourite } = useVehicles();

  const handleGoBack = () => {
    navigate(ROUTES.LISTING);
  };

  const vehicle = useMemo(() => {
    if (!vehicleId) {
      return;
    }

    return getVehicleById(vehicleId);
  }, [getVehicleById, vehicleId]);

  const auctionDate = moment(vehicle?.auctionDateTime, "YYYY/MM/DD HH:mm:ss");
  const formatted = auctionDate.format("DD MMM YYYY, HH:mm");

  const timeToAuction = getRelativeTime(new Date(formatted));

  const bidFormatter = new Intl.NumberFormat("en-GB", {
    style: "currency",
    currency: "GBP",
  });

  if (!vehicle) {
    return (
      <Grid container spacing="16px" data-testid="error-container">
        <Grid item xs={12} sm={12} md={12} textAlign="center">
          <Typography variant="body1" color={defaultColors.primaryMain}>
            Something went wrong. Please try again later
          </Typography>
        </Grid>
      </Grid>
    );
  }

  return (
    <Grid container spacing="16px">
      <Grid item xs={12} sm={12} md={12}>
        <Button
          data-testid="back-button"
          startIcon={<ArrowBackIcon />}
          variant="text"
          sx={{ color: defaultColors.neutralDarkerGrey }}
          onClick={handleGoBack}>
          Back to list
        </Button>
      </Grid>
      <Grid item xs={12} sm={12} md={12} data-testid="vehicle-main-info">
        <Box
          display="flex"
          position="relative"
          padding="16px"
          borderRadius="30px 4px"
          gap="16px"
          sx={{
            backgroundColor: defaultColors.neutralWhite,
            border: "1px solid #f2f2f2",
          }}>
          <Grid container spacing="16px">
            <Grid item xs={12} sm={7} md={7} display="flex" flexDirection="column" gap="16px">
              <Box display="flex" justifyContent="center" alignItems="center" padding="0 32px">
                <img
                  src={emptyPhoto}
                  alt="Car"
                  style={{
                    width: "100%",
                    maxWidth: "300px",
                    objectFit: "cover",
                    objectPosition: "center",
                    borderRadius: "12px",
                    display: "block",
                    opacity: 0.7,
                  }}
                />
              </Box>
              <Box
                display="flex"
                flexWrap="wrap"
                justifyContent="center"
                alignItems="center"
                gap="12px"
                padding="0 32px">
                <img
                  src={emptyPhoto}
                  alt="Car"
                  style={{
                    width: "110px",
                    objectFit: "cover",
                    objectPosition: "center",
                    borderRadius: "12px",
                    display: "block",
                    opacity: 0.7,
                  }}
                />
                <img
                  src={emptyPhoto}
                  alt="Car"
                  style={{
                    width: "110px",
                    objectFit: "cover",
                    objectPosition: "center",
                    borderRadius: "12px",
                    display: "block",
                    opacity: 0.7,
                  }}
                />
                <img
                  src={emptyPhoto}
                  alt="Car"
                  style={{
                    width: "110px",
                    objectFit: "cover",
                    objectPosition: "center",
                    borderRadius: "12px",
                    display: "block",
                    opacity: 0.7,
                  }}
                />
                <img
                  src={emptyPhoto}
                  alt="Car"
                  style={{
                    width: "110px",
                    objectFit: "cover",
                    objectPosition: "center",
                    borderRadius: "12px",
                    display: "block",
                    opacity: 0.7,
                  }}
                />
              </Box>
            </Grid>
            <Grid item xs={12} sm={5} md={5}>
              <IconButton
                onClick={() => toggleFavourite(vehicle.id)}
                sx={{
                  position: "absolute",
                  top: 8,
                  right: 8,
                  backgroundColor: "white",
                  "&:hover": {
                    backgroundColor: "grey.100",
                  },
                }}>
                {vehicle.favourite ? (
                  <StarOutlinedIcon color="warning" />
                ) : (
                  <StarBorderOutlinedIcon />
                )}
              </IconButton>
              <Box display="flex" flexDirection="column" gap="24px">
                <Typography variant="h5" color={defaultColors.primaryMain} fontSize="20px">
                  {vehicle.make} - {vehicle.model}
                </Typography>
                <LabelledText title="Fuel" text={vehicle.fuel} />
                <LabelledText title="Vehicle Year" text={`${vehicle.year}`} />
                <LabelledText
                  title="Vehicle Mileage"
                  text={`${new Intl.NumberFormat("pt-PT").format(vehicle.mileage) ?? 0} Miles`}
                />
                <LabelledText
                  title="Starting Bid"
                  text={bidFormatter.format(vehicle.startingBid)}
                />
                <LabelledText title="Auction Date" text={formatted} moreInfo={timeToAuction} />
                <Button
                  variant="contained"
                  color="primary"
                  onClick={() => {}}
                  disabled
                  sx={{
                    boxShadow: "none",
                    backgroundColor: defaultColors.primaryMain,
                    ":hover": {
                      boxShadow: "none",
                    },
                  }}>
                  Bid now
                </Button>
              </Box>
            </Grid>
          </Grid>
        </Box>
      </Grid>
      <Grid
        item
        xs={12}
        sm={12}
        md={12}
        data-testid="vehicle-additional-info"
        display="flex"
        flexDirection="column"
        gap="16px">
        {vehicle?.details?.specification && (
          <Accordion
            id={`${vehicle.id}-specification`}
            header={
              <Typography variant="h5" color={defaultColors.primaryLighter}>
                Additional information
              </Typography>
            }>
            <Grid container>
              <Grid item xs={12} sm={4} md={4}>
                <LabelledText
                  title="Vehicle Type"
                  text={vehicle?.details?.specification?.vehicleType}
                />
              </Grid>
              <Grid item xs={12} sm={4} md={4}>
                <LabelledText
                  title="Vehicle Colour"
                  text={vehicle?.details?.specification?.colour}
                />
              </Grid>
              <Grid item xs={12} sm={4} md={4}>
                <LabelledText
                  title="Vehicle Transmission"
                  text={vehicle?.details?.specification?.transmission}
                />
              </Grid>
              <Grid item xs={12} sm={4} md={4}>
                <LabelledText
                  title="No of Doors"
                  text={`${vehicle?.details?.specification?.numberOfDoors}`}
                />
              </Grid>
              <Grid item xs={12} sm={4} md={4}>
                <LabelledText
                  title="No of Keys"
                  text={`${vehicle?.details?.specification?.numberOfKeys}`}
                />
              </Grid>
              <Grid item xs={12} sm={4} md={4}>
                <LabelledText
                  title="CO2 Emissions"
                  text={vehicle?.details?.specification?.co2Emissions}
                />
              </Grid>
              <Grid item xs={12} sm={4} md={4}>
                <LabelledText
                  title="NOX Emissions"
                  text={`${vehicle?.details?.specification?.noxEmissions}`}
                />
              </Grid>
            </Grid>
          </Accordion>
        )}
        {vehicle?.details?.ownership && (
          <Accordion
            id={`${vehicle.id}-ownership`}
            header={
              <Typography variant="h5" color={defaultColors.primaryLighter}>
                Ownership information
              </Typography>
            }>
            <Grid container>
              <Grid item xs={12} sm={4} md={4}>
                <LabelledText title="Logbook" text={vehicle?.details?.ownership?.logBook} />
              </Grid>
              <Grid item xs={12} sm={4} md={4}>
                <LabelledText
                  title="No of Owners"
                  text={`${vehicle?.details?.ownership?.numberOfOwners}`}
                />
              </Grid>
              <Grid item xs={12} sm={4} md={4}>
                <LabelledText
                  title="Date of Registration"
                  text={moment(vehicle?.details?.ownership?.dateOfRegistration)?.format(
                    "DD MMM YYYY, HH:mm",
                  )}
                />
              </Grid>
            </Grid>
          </Accordion>
        )}
        {vehicle?.details?.equipment?.length && (
          <Accordion
            id={`${vehicle.id}-equipment`}
            header={
              <Typography variant="h5" color={defaultColors.primaryLighter}>
                Equipment information
              </Typography>
            }>
            <Box display="flex" flexDirection="column" gap="12px">
              {vehicle.details.equipment.map((item) => (
                <Typography
                  variant="body1"
                  color={defaultColors.neutralDarkGrey}
                  sx={{ textTransform: "capitalize" }}>
                  {`• ${item};`}
                </Typography>
              ))}
            </Box>
          </Accordion>
        )}
      </Grid>
    </Grid>
  );
};

export default VehicleDetails;
