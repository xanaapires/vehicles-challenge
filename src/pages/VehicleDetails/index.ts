export { default as VehicleDetails } from "./VehicleDetails";
