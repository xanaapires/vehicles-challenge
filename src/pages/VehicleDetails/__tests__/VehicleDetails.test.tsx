import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter, useParams } from "react-router-dom";
import VehicleDetails from "../VehicleDetails";
import { vehiclesData } from "../../../assets/vehiclesData";
import { defaultStateMock } from "../../../utils/hookFixtures";
import { ROUTES } from "../../../utils/routes";

const mockedNavigate = jest.fn();
const mockUseVehicles = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedNavigate,
  useParams: jest.fn(),
}));

jest.mock("../../../hooks/useVehicles", () => ({
  useVehicles: () => mockUseVehicles(),
}));

const renderComponent = () => {
  return render(
    <MemoryRouter>
      <VehicleDetails />
    </MemoryRouter>,
  );
};

describe("VehicleDetails", () => {
  beforeAll(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date("2025-08-04T00:00:00Z"));
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  test("Renders VehicleDetails correctly", () => {
    mockUseVehicles.mockReturnValue({ ...defaultStateMock, getVehicleById: () => vehiclesData[1] });
    (useParams as jest.Mock).mockReturnValue({ vehicleId: "1" });

    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("vehicle-main-info")).toBeVisible();
    expect(screen.queryAllByTestId("accordion")).toHaveLength(3);
  });

  test("Handles unexpected error correctly", () => {
    mockUseVehicles.mockReturnValue({ ...defaultStateMock, getVehicleById: () => null });
    (useParams as jest.Mock).mockReturnValue({ vehicleId: "" });

    renderComponent();

    expect(screen.getByTestId("error-container")).toBeVisible();
  });

  test("Handles back to listing page", () => {
    mockUseVehicles.mockReturnValue({ ...defaultStateMock, getVehicleById: () => vehiclesData[1] });
    (useParams as jest.Mock).mockReturnValue({ vehicleId: "1" });

    renderComponent();

    expect(screen.getByTestId("vehicle-main-info")).toBeVisible();
    expect(screen.getByTestId("back-button")).toBeVisible();

    fireEvent.click(screen.getByTestId("back-button"));

    expect(mockedNavigate).toHaveBeenCalledWith(ROUTES.LISTING);
  });
});
