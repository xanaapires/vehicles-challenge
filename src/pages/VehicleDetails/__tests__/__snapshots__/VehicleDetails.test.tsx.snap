// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleDetails Renders VehicleDetails correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-vq5nh1-MuiButtonBase-root-MuiButton-root"
        data-testid="back-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"
            />
          </svg>
        </span>
        Back to list
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
      data-testid="vehicle-main-info"
    >
      <div
        class="MuiBox-root css-10trl62"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-7 MuiGrid-grid-md-7 css-1rh321n-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-qixr6d"
            >
              <img
                alt="Car"
                src="empty_photo.jpg"
                style="width: 100%; max-width: 300px; object-fit: cover; object-position: center; border-radius: 12px; display: block; opacity: 0.7;"
              />
            </div>
            <div
              class="MuiBox-root css-zlnkop"
            >
              <img
                alt="Car"
                src="empty_photo.jpg"
                style="width: 110px; object-fit: cover; object-position: center; border-radius: 12px; display: block; opacity: 0.7;"
              />
              <img
                alt="Car"
                src="empty_photo.jpg"
                style="width: 110px; object-fit: cover; object-position: center; border-radius: 12px; display: block; opacity: 0.7;"
              />
              <img
                alt="Car"
                src="empty_photo.jpg"
                style="width: 110px; object-fit: cover; object-position: center; border-radius: 12px; display: block; opacity: 0.7;"
              />
              <img
                alt="Car"
                src="empty_photo.jpg"
                style="width: 110px; object-fit: cover; object-position: center; border-radius: 12px; display: block; opacity: 0.7;"
              />
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-5 css-11lxi55-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1u2l4gi-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="StarBorderOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiBox-root css-1i43dhb"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-o8v361-MuiTypography-root"
              >
                Ford
                 - 
                Fiesta
              </h5>
              <div
                class="MuiBox-root css-0"
              >
                <span
                  class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                >
                  Fuel
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                >
                  petrol
                </p>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <span
                  class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                >
                  Vehicle Year
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                >
                  2022
                </p>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <span
                  class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                >
                  Vehicle Mileage
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                >
                  9084 Miles
                </p>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <span
                  class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                >
                  Starting Bid
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                >
                  Â£15,000.00
                </p>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <span
                  class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                >
                  Auction Date
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                >
                  15 Apr 2024, 09:00
                </p>
                <span
                  class="MuiTypography-root MuiTypography-body3 css-jouwp7-MuiTypography-root"
                >
                  (
                  475 day(s) and 16 hour(s) ago
                  )
                </span>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-9wvbqu-MuiButtonBase-root-MuiButton-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                Bid now
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 css-ok2iup-MuiGrid-root"
      data-testid="vehicle-additional-info"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1b2vurp-MuiPaper-root-MuiAccordion-root"
        data-testid="accordion"
      >
        <div
          aria-controls="accordion-details-undefined-specification"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1e7l4tw-MuiButtonBase-root-MuiAccordionSummary-root"
          id="accordion-summary-undefined-specification"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-kcq9hy-MuiTypography-root"
            >
              Additional information
            </h5>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="accordion-summary-undefined-specification"
                class="MuiAccordion-region"
                id="accordion-details-undefined-specification"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-1is2or-MuiAccordionDetails-root"
                  id="accordion-details-undefined-specification"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-wg4xta-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                        >
                          Vehicle Type
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                        >
                          Car
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-wg4xta-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                        >
                          Vehicle Colour
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                        >
                          RED
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-wg4xta-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                        >
                          Vehicle Transmission
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                        >
                          Manual
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-wg4xta-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                        >
                          No of Doors
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                        >
                          3
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-wg4xta-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                        >
                          No of Keys
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                        >
                          2
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-wg4xta-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                        >
                          CO2 Emissions
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                        >
                          139 g/km
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-wg4xta-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                        >
                          NOX Emissions
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                        >
                          230
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1b2vurp-MuiPaper-root-MuiAccordion-root"
        data-testid="accordion"
      >
        <div
          aria-controls="accordion-details-undefined-ownership"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1e7l4tw-MuiButtonBase-root-MuiAccordionSummary-root"
          id="accordion-summary-undefined-ownership"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-kcq9hy-MuiTypography-root"
            >
              Ownership information
            </h5>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="accordion-summary-undefined-ownership"
                class="MuiAccordion-region"
                id="accordion-details-undefined-ownership"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-1is2or-MuiAccordionDetails-root"
                  id="accordion-details-undefined-ownership"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-wg4xta-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                        >
                          Logbook
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                        >
                          Present
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-wg4xta-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                        >
                          No of Owners
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                        >
                          8
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-wg4xta-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-body3 css-1p3ip6s-MuiTypography-root"
                        >
                          Date of Registration
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                        >
                          31 Mar 2015, 09:00
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1b2vurp-MuiPaper-root-MuiAccordion-root"
        data-testid="accordion"
      >
        <div
          aria-controls="accordion-details-undefined-equipment"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1e7l4tw-MuiButtonBase-root-MuiAccordionSummary-root"
          id="accordion-summary-undefined-equipment"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-kcq9hy-MuiTypography-root"
            >
              Equipment information
            </h5>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="accordion-summary-undefined-equipment"
                class="MuiAccordion-region"
                id="accordion-details-undefined-equipment"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-1is2or-MuiAccordionDetails-root"
                  id="accordion-details-undefined-equipment"
                >
                  <div
                    class="MuiBox-root css-1hdbc19"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                    >
                      â¢ Air Conditioning;
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                    >
                      â¢ Tyre Inflation Kit;
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                    >
                      â¢ Photocopy of V5 Present;
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                    >
                      â¢ Navigation HDD;
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                    >
                      â¢ 17 Alloy Wheels;
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                    >
                      â¢ Engine Mods/Upgrades;
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-3uxmf5-MuiTypography-root"
                    >
                      â¢ Modifd/Added Body Parts;
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
